src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutSubItem.cs

```diff
@@ -1,47 +1,895 @@
+						shouldHandleEvent = true;
+					}
+				}
+			}
+
+			m_menuHelper.OnKeyDown(args);
+			args.Handled = !!shouldHandleEvent;
+		}
+
+
+		protected override void OnKeyUp(KeyRoutedEventArgs args)
+		{
+			base.OnKeyUp(args);
+
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: OnKeyUp.", this));
+#endif // MFSI_DEBUG
+
+			m_menuHelper.OnKeyUp(args);
+		}
+
+		// Ensure the creating the popup and menu presenter to show the 
+		void EnsurePopupAndPresenter()
+		{
+#if MFSI_DEBUG
+			IGNOREHR(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "MFSI[0x%p]: EnsurePopupAndPresenter.", this));
+#endif // MFSI_DEBUG
+
+			if (m_tpPopup == null)
+			{
+				MenuFlyoutPresenter spParentMenuFlyoutPresenter = null;
+				Control spPresenter;
+				UIElement spPresenterAsUI;
+				FrameworkElement spPresenterAsFE;
+				Popup spPopup;
+
+				spPopup = new Popup();
+				spPopup.IsSubMenu = true;
+				spPopup.IsLightDismissEnabled = false;
+
+#if false // UNO TODO Windowed Popup is not available
+				spParentMenuFlyoutPresenter = GetParentMenuFlyoutPresenter();
+				if (spParentMenuFlyoutPresenter != null)
+				{
+					spParentMenuFlyout = spParentMenuFlyoutPresenter.GetParentMenuFlyout();
+					// Set the windowed Popup if the MenuFlyout is set the windowed Popup
+					if (spParentMenuFlyout && spParentMenuFlyout.IsWindowedPopup())
+					{
+						ASSERT((CPopup*)(spPopup as Popup.GetHandle()).DoesPlatformSupportWindowedPopup(DXamlCore.GetCurrent().GetHandle()));
+
+						((CPopup*)(spPopup as Popup.GetHandle()).SetIsWindowed());
+
+						// Ensure the sub menu is the windowed Popup
+						ASSERT((CPopup*)(spPopup as Popup.GetHandle()).IsWindowed());
+
+						xaml.IXamlRoot xamlRoot = XamlRoot.GetForElementStatic(spParentMenuFlyoutPresenter);
+						if (xamlRoot)
+						{
+							(spPopup as Popup.XamlRoot = xamlRoot);
+						}
+					}
+				}
+#endif
+
+				spPresenter = CreateSubPresenter();
+				spPresenterAsUI = spPresenter;
+
+				if (spParentMenuFlyoutPresenter != null)
+				{
+					int parentDepth = spParentMenuFlyoutPresenter.GetDepth();
+					(spPresenter as MenuFlyoutPresenter).SetDepth(parentDepth + 1);
+				}
+
+				spPopup.Child = spPresenterAsUI as FrameworkElement;
+
+				m_tpPresenter = spPresenter;
+				m_tpPopup = spPopup;
+
+				((ItemsControl)(m_tpPresenter)).ItemsSource = m_tpItems;
+
+				spPresenterAsFE = spPresenter;
+
+				spPresenterAsFE.SizeChanged += OnPresenterSizeChanged;
+				m_epPresenterSizeChangedHandler = Disposable.Create(() => spPresenterAsFE.SizeChanged -= OnPresenterSizeChanged);
+
+				m_menuHelper.SetSubMenuPresenter(spPresenter as Control);
+			}
+		}
+
+		void ForwardPresenterProperties(
+			MenuFlyout pOwnerMenuFlyout,
+			MenuFlyoutPresenter pParentMenuFlyoutPresenter,
+			MenuFlyoutPresenter pSubMenuFlyoutPresenter)
+		{
+			Style spStyle;
+			ElementTheme parentPresenterTheme;
+			object spDataContext;
+			FlowDirection flowDirection = FlowDirection.LeftToRight;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'flowDirection'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)